crates/forge-runner/src/gas.rs

```diff
@@ -1,33 +1,44 @@
 use std::collections::HashMap;
 
+use blockifier::fee::eth_gas_constants;
+use blockifier::fee::fee_utils::calculate_tx_l1_gas_usage;
 use blockifier::fee::os_resources::OS_RESOURCES;
 use blockifier::fee::os_usage::get_additional_os_resources;
 use blockifier::transaction::transaction_types::TransactionType;
 use blockifier::{
     abi::constants, block_context::BlockContext, execution::entry_point::ExecutionResources,
-    fee::fee_utils::calculate_l1_gas_by_vm_usage, transaction::objects::ResourcesMapping,
+    transaction::objects::ResourcesMapping,
 };
 use cairo_vm::vm::runners::cairo_runner::ExecutionResources as VmExecutionResources;
 
 #[must_use]
 pub fn gas_from_execution_resources(
     block_context: &BlockContext,
     resources: &ExecutionResources,
-) -> f64 {
+    onchain_data_segment_len: usize,
+) -> u128 {
     let total_vm_usage = get_total_vm_usage(resources);
-    let resource_mapping = vm_execution_resources_to_resource_mapping(&total_vm_usage);
-    calculate_l1_gas_by_vm_usage(block_context, &resource_mapping)
+    let resource_mapping =
+        vm_execution_resources_to_resource_mapping(&total_vm_usage, onchain_data_segment_len);
+    calculate_tx_l1_gas_usage(&resource_mapping, block_context)
         .expect("Calculating gas failed, some resources were not included.")
 }
 
 #[must_use]
 fn vm_execution_resources_to_resource_mapping(
     execution_resources: &VmExecutionResources,
+    onchain_data_segment_len: usize,
 ) -> ResourcesMapping {
-    let mut map = HashMap::from([(
-        constants::N_STEPS_RESOURCE.to_string(),
-        execution_resources.n_steps,
-    )]);
+    let mut map = HashMap::from([
+        (
+            constants::N_STEPS_RESOURCE.to_string(),
+            execution_resources.n_steps,
+        ),
+        (
+            constants::GAS_USAGE.to_string(),
+            onchain_data_segment_len * eth_gas_constants::SHARP_GAS_PER_MEMORY_WORD,
```

Comment:
Blockifier says that [this value](https://github.com/starkware-libs/blockifier/blob/b10f502c31b30db90bed6469b5f8a41159091fa2/crates/blockifier/src/fee/eth_gas_constants.rs#L4) is not accurate, so computed gas can be wrong.

Reply:
We should mark this as approximate value when printing `gas: ~432134` or sth like that cc @cptartur

Reply:
I think we should print what we assume the cost os gas per memory word on L1 is

Reply:
> I think we should print what we assume the cost of gas per memory word on L1 is

What do you mean by that? I don't think we should complicated this feature anymore than just outputting a total estimated value.